### ISSUE TITLE
unable to initialize policy plugin issue [Ubuntu 16.04]

### ISSUE THREAD

**Issue by icon256** [factoriommo/factorio-multienv-ctl#18]
Attempting to upgrade to factorio-multienv-ctl (0.16.7) to run FENV= /etc/factorio/factorio, it spits out this error:  _"unable to initialize policy plugin"_.  Running Ubuntu 16.04 originally on version factorio-multienv-ctl (0.15.28).  

I even attempted "upgraded", by just writing over the old changed files while preserving my config items within the factorio-multienv-ctl folder from (0.15.28).  I believe my etc, opt, var folders are correctly setup as I run factorio-multienv-ctl (0.15.28) works just fine; I preserve the original folder with working config as reverted to running this for now until I figure out the direct issue.  I don't have a user permission issue using on this version and runs as designed.  Doubled check for all apt-updates and apt-upgrades completed.

After some research and reading of the changelog, I think something broke for ubuntu since the version below:
factorio-multienv-ctl (0.15.31-1) unstable; urgency=low
  * Patch to use sudo instead of su to provide credential caching when run as
    user.

I could be wrong, but I think its the variable that changed down the line, $FUSERNAME:$USERGROUP.
Would one be able to give me a clue on this issue or what may have changed that I need to modify on my system or Ubuntu.

**Comment by psihius** [comment]
@maikelwever could you take a look at this?

@icon256 can you try just removing the package and installing a version you want?
It's quite far back now and we had issues of our own when using `su`, we had bug reports from other people and that was the fix for it.

**Comment by maikelwever** [comment]
As long as this cannot be reliably reproduced (and there are no instruction for this), it's not worth even trying.

@icon256 does your sudo even work at all?

**Comment by icon256** [comment]
Hi guys,

Thanks for replying...Sorry I haven't seen this (during the holidays, USA) and replied sooner.

Just getting back into things.  Sudo works fine,

I'm able to reproduce this if need.  After a good amount of trial and error, I figured out that it's insisting on using "factorio" as a user, despite configuring init.conf to use a known user; in this case "ubuntu".  As stated, in previous versions below factorio-multienv-ctl (0.15.31-1), it seems to respect my init.conf user names in the various environmental variables (FENV=game1).  Long story short, I don't use factorio as a linux user, and never have.

I have completely wiped out and downloaded the latest factorio-multienv-ctl clone, actually today (0.16.15) as well, and attempt to configure as my previous version.  The only work around I found is to change line 35/36 in factorio file to:

    if [ -z "${FUSERNAME}" ]; then
      FUSERNAME=ubuntu

So in other words, where should I technically change that to have it respect my linux user names.  I don't recall having to changing it anywhere else other than my usual respective FENV= environment directories init.conf files for line 8/9:

    USERGROUP=ubuntu
    USERNAME=ubuntu

Thanks to the programmers of great script to multi-host without eating 7x multi-instances of factorio!

**Comment by psihius** [comment]
@icon256 thanks for the info.

Well, the package itself was done so it creates `factorio` user and group, so I think we didn't really try and test how it goes with other users... :)

I'll let @maikelwever to review this, as he knows better and it looks kind'a strange that ubuntu/ubuntu does not work for some reason in the script.
